### PR TITLE
cli: config: add docs for --show-origin

### PR DESCRIPTION
**Pull request by mrstegeman**
Support the --show-origin option for config, which, similar to git,
prefixes each config option with the source file it originated from.

Per iterative/dvc#5119
Matches iterative/dvc#5126

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Review by shcheklein — APPROVED** [review]

File: content/docs/command-reference/config.md [hunk 1]
```diff
@@ -67,6 +67,9 @@ multiple projects and users, respectively:
 
 - `-l`, `--list` - lists all defined config values.
 
+- `--show-origin` - when listing or getting config values, also show the file
+  the config value is saved in.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/config.md:71]: Copy edit suggestion:
```suggestion
- `--show-origin` - when listing or getting config options, also show the
  location of the config file where each option `value` is found.
```

Comment (jorgeorpinel) [reply content/docs/command-reference/config.md:71]: What does "getting" refer to here?

Comment (jorgeorpinel) [reply content/docs/command-reference/config.md:71]: > What does "getting" refer to here though?

Never mind, I got it.

**Comment by jorgeorpinel** [comment]
I think that it would be great to mention the option somewhere in the description around where the config file location tables are — probably as a note (md block quote).

File: content/docs/command-reference/config.md (hunk 1 above)
Comment (mrstegeman) [reply content/docs/command-reference/config.md]: Done.

**Comment by mrstegeman** [comment]
> I think that it would be great to mention the option somewhere in the description around where the config file location tables are — probably as a note (md block quote).

Done.

File: content/docs/command-reference/config.md [hunk 2]
```diff
@@ -42,6 +42,9 @@ multiple projects and users, respectively:
 | `--global` | 3        | `$HOME/Library/Application\ Support/dvc` | `$HOME/.config/dvc/config` | `%LocalAppData%\iterative\dvc\config`                     |
 | `--system` | 4        | `/Library/Application\ Support/dvc`      | `/etc/dvc/config`          | `%AllUsersProfile%\Application Data\iterative\dvc\config` |
 
+> **Note:** The `--show-origin` flag can show you where a given config option
+> value is currently stored.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/config.md:46]: Stylistic copy edit, committing:
```suggestion
> Note that the `--show-origin` flag can show you where a given config option
> `value` is currently stored.
```

**Review by jorgeorpinel — APPROVED** [review]
Woah that was fast. Thanks @mrstegeman! LGTM

**Pull request merged**